packages/dev/serializers/src/USD/USDSerializer.ts

```diff
@@ -0,0 +1,694 @@
+
+            inputs.push(`${pad}float inputs:metallic.connect = </Materials/Material_${material.uniqueId}/Texture_${maps.metallic.uniqueId}_metallic.outputs:b>`);
+            samplers.push(buildTexture(maps.metallic, "metallic"));
+
+            inputs.push(`${pad}float inputs:occlusion.connect = </Materials/Material_${material.uniqueId}/Texture_${maps.ao.uniqueId}_occlusion.outputs:r>`);
+            samplers.push(buildTexture(maps.ao, "occlusion"));
+        } else {
+            inputs.push(`${pad}float inputs:roughness = ${material.roughness}`);
+            inputs.push(`${pad}float inputs:metallic = ${material.metallic}`);
+        }
+
+        if (material.opacityTexture !== null) {
+            inputs.push(`${pad}float inputs:opacity.connect = </Materials/Material_${material.uniqueId}/Texture_${material.opacityTexture.uniqueId}_opacity.outputs:r>`);
+            inputs.push(`${pad}float inputs:opacityThreshold = 0.0001`);
+            samplers.push(buildTexture(material.opacityTexture as Texture, "opacity"));
+        } else {
+            inputs.push(`${pad}float inputs:opacity = ${material.alpha}`);
+        }
+
+        if (material.isMetallicWorkflow()) {
+            inputs.push(`${pad}float inputs:clearcoat = ${0.0}`);
+            inputs.push(`${pad}float inputs:clearcoatRoughness = ${0.1}`);
+            inputs.push(`${pad}float inputs:ior = ${material.metallicF0Factor}`);
+        }
+
+        return `
+        def Material "Material_${material.uniqueId}"
+        {
+            def Shader "PreviewSurface"
+            {
+                uniform token info:id = "UsdPreviewSurface"
+    ${inputs.join("\n")}
+                int inputs:useSpecularWorkflow = 0
+                token outputs:surface
+            }
+    
+            token outputs:surface.connect = </Materials/Material_${material.uniqueId}/PreviewSurface.outputs:surface>
+    
+    ${samplers.join("\n")}
+        }
+    `;
+    }
+
+    private static _BuildColor(color: Color3) {
+        return `(${color.r}, ${color.g}, ${color.b})`;
+    }
+
+    private static _BuildColor4(color: Color3) {
+        return `(${color.r}, ${color.g}, ${color.b}, 1.0)`;
+    }
+
+    private static _BuildVector2(vector: Vector2) {
+        return `(${vector.x}, ${vector.y})`;
+    }
+
+    private static _BuildUSDFileAsString(dataToInsert: string) {
+        let output = USDExport._BuildHeader();
+        output += dataToInsert;
+        //output += USDExport.BuildFooter();
+        return strToU8(output);
+    }
+
+    /**
+     * Converts a Babylon Texture to an HTML Image Element.
+     * @param texture can be a Texture or DynamicTexture
+     * @returns a promise with the HTML Image Element or the DynamicTexture directly.
+     */
+    private static async _TextureToImage(texture: Texture | DynamicTexture): Promise<HTMLImageElement> {
+        if (texture.getClassName() === "DynamicTexture") {
+            return texture as any;

```

Comment:
Why not returning `Promise<HTMLImageElement | DynamicTexture>` and avoiding the `any` cast?